docs/strategies/hedging.md

```diff
@@ -440,6 +440,40 @@ sequenceDiagram
     P->>C: Propagates failure (R3)
 ```
 
+## Hedging and resilience context
+
+The hedging strategy supports the concurrent execution and cancellation of multiple hedged actions. To safeguard against concurrency issues when these actions access the same `ResilienceContext`, the hedging strategy ensures that each hedged action has its unique `ResilienceContext`. The hedging strategy distinguishes between two types of contexts:
+
+- **Primary context**: The original resilience context that the hedging strategy receives.
+- **Action context**: A deep copy of the primary context with distinct cancellation token for a single hedged action.
+
+Here's the flow:
```

Comment:
I was thinking whether it make sense to add a mermaid sequence diagram here as well. + also including `OnHedging` and `ActionGenerator`. 

What do you think?

Reply:
Sure, it could be useful, let's do it in follow-up. 

Can you give us a hand here?

Reply:
Yepp, sure 👍